public/static/docs/api-reference/open.md

```diff
@@ -0,0 +1,178 @@
+  the target data. If not supplied, the cache directory is tried first for local
+  projects; The default remote of `repo` is tried otherwise.
+
+  A `dvc.exceptions.NoRemoteError` is raised if no `remote` is found.
+
+- `mode` - (optional) mirrors the namesake parameter in builtin
+  [`open()`](https://docs.python.org/3/library/functions.html#open). Defaults to
+  `"r"` (read).
+
+- `encoding` - (optional) used to decode contents to a string. Mirrors the
+  namesake parameter in builtin `open()`. Defaults to `"utf-8"`.
+
+## Example: Process XML file from an external DVC repository
+
+```py
+from xml.dom.minidom import parse
+
+import dvc.api
+
+with dvc.api.open(
+        'get-started/data.xml',
+        repo='https://github.com/iterative/dataset-registry'
+        ) as fd:
+    xmldom = parse(fd)
+    # ... Process elements
+```
+
+> See `dvc.api.read()` for a shorthand way to read the contents of a tracked
+> file.
+
+## Example: Use a file from the local cache
+
+In this case we don't supply a `repo` value. DVC will walk up the current
+working directory tree to find the <abbr>DVC project</abbr>:
+
+```py
+import dvc.api
+
+with dvc.api.open('data/nlp/words.txt') as fd:
+    for word in fd:
+        # ... Process words
+```
+
+DVC will look for `data/nlp/words.txt` in the local cache of the
+<abbr>project</abbr>. (If it's not found there, the default
+[remote](/doc/command-reference/remote) will be tried.)
+
+To specify the file encoding of a text file:
+
+```py
+with dvc.api.open('data/nlp/words.txt', encoding='utf-8') as fd:
+    # ...
+```
+
+## Example: Process CSV file from a private repository
+
+For this we'll have to use the SSH URL to the Git repo (assuming the local
+[SSH credentials](https://help.github.com/en/github/authenticating-to-github/connecting-to-github-with-ssh)
+are configured locally):
+
+```py
+import csv
+import dvc.api
+
+with dvc.api.open(
+        'sea_ice.csv',
+        repo='git@github.com:iterative/df_sea_ice_no_header.git'
+        ) as fd:
+    reader = csv.reader(fd)
+    for row in reader:
+        # ... Process columns
+```
+
+> Note that we're using an SSH Git URL for the `repo` argument above.
+
+## Example: Stream file from a specific remote
+
+Sometimes we may want to choose the [remote](/doc/command-reference/remote) data
+source, for example to ensure that file streaming is enabled (as only certain
+remote storage types support streaming). This can be done by providing a
+`remote` argument:
+
+```py
+import dvc.api
+
+with open(
+        'activity.log',
+        repo='location/of/dvc/project',
+        remote='my-s3-bucket'
+        ) as fd:
+    for line in fd:
+        match = re.search(r'user=(\w+)', line)
+        # ...
+```
+
+## Example: Unserialize and employ a binary model
+
+```py
+import pickle
+import dvc.api
+
+with dvc.api.open('model.pkl', repo='...') as fd:
+    model = pickle.load(fd)
+    # ... Use model
+```
+
+> For a faster way to perform a similar example, please see the
```

Comment:
let's just show that code snippet here?

Reply:
Well, this note was already removed. But a similar one existed for **Example: Use artifacts from online DVC repositories**. I've added a full read() code snippet in it now. Please see 35674e36